src/test/standalone/notifications/pythonExtensionRestartNotification.vscode.test.ts

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { assert } from 'chai';
+import * as sinon from 'sinon';
+import { EventEmitter } from 'vscode';
+import { IApplicationShell, IVSCodeNotebook } from '../../../platform/common/application/types';
+import { IDisposable, IDisposableRegistry } from '../../../platform/common/types';
+import { sleep } from '../../core';
+import { DataScience } from '../../../platform/common/utils/localize';
+import { IExtensionTestApi, initialize, startJupyterServer } from '../../common';
+import {
+    closeNotebooksAndCleanUpAfterTests,
+    createEmptyPythonNotebook,
+    hijackPrompt,
+    insertCodeCell,
+    runCell,
+    waitForTextOutput,
+    WindowPromptStub
+} from '../../datascience/notebook/helper';
+import { IPythonExtensionChecker } from '../../../platform/api/types';
+import { IServiceContainer } from '../../../platform/ioc/types';
+import { PythonExtensionRestartNotification } from '../../../standalone/notification/pythonExtensionRestartNotification';
+import { IKernelProvider } from '../../../kernels/types';
+import { noop } from '../../../platform/common/utils/misc';
+
+suite('Python Extension Restart Notification', () => {
+    let api: IExtensionTestApi;
+    let extensionChecker: IPythonExtensionChecker;
+    let vscodeNotebook: IVSCodeNotebook;
+    let hijackedMessage: WindowPromptStub;
+    let fakeInstalledEventEmitter: EventEmitter<'installed' | 'uninstalled'>;
+    const disposables: IDisposable[] = [];
+    suiteSetup(async function () {
+        this.timeout(120_000);
+        api = await initialize();
+        await closeNotebooksAndCleanUpAfterTests();
+        vscodeNotebook = api.serviceContainer.get<IVSCodeNotebook>(IVSCodeNotebook);
+    });
+    setup(async function () {
+        api = await initialize();
+        extensionChecker = api.serviceContainer.get<IPythonExtensionChecker>(IPythonExtensionChecker);
+        sinon.restore();
+
+        // Stub out a fake event emitter for the extension checker to simulate install
+        fakeInstalledEventEmitter = new EventEmitter<'installed' | 'uninstalled'>();
+        sinon.stub(extensionChecker, 'onPythonExtensionInstallationStatusChanged').get(() => {
+            return fakeInstalledEventEmitter.event;
+        });
+
+        // Hijack the info message prompt
+        hijackedMessage = await hijackPrompt('showInformationMessage', {
+            contains: DataScience.pythonExtensionInstalled()
+        });
+
+        // Create an instance of the restart notification with the stubbed checker
+        const notifier = createService(api.serviceContainer);
```

Comment:
Test was a little bit hard to write, since I didn't want to actually install / uninstall python. Poked around at a few of Don's bag of tricks and I think this works. Stub out the extension installed event from the checker, create a new instance of the class (which picks up the stubbed event), and see if that notifies or not.